src/content/blog/val-town-newsletter-22.mdx

```diff
@@ -0,0 +1,308 @@
+import BlobAdmin from "./val-town-newsletter-22/BlobAdmin.png";
+
+Val Town is growing up! We're working on new features and infra that will fundamentally unlock the scope of what you can build in Val Town. We have two major efforts running in parallel: projects and scale.
+
+_Projects_ are containers for multiple vals, folders, and files – enabling you to create infinitely complex real-world apps – instead of being limited to a single long val. Internal to a project, vals will use relative imports (ie `./foo/bar`), so they can be forked as a unit, and merged back in as a unit. Projects will also improve Townie's ability to help write your vals in more modular files – so it doesn't have to generate and re-generate all the code on every change.
+
+_Scale_ will enable you to run production workloads in Val Town, with usage-based pricing, custom alerts, and caps.
+
+We hope to have both of these efforts live in the coming months. In the meantime, we've shipped many other features and improvements, including a dozen Townie upgrades, scoped API permissions, a partnership with Fal, and more.
+
+[Justin Bennett](https://just-be.dev) has joined our team! He is well known for the [DevtoolsFM podcast](https://www.devtools.fm/), and was most recently at [The Recurse Center](https://www.recurse.com/), [Membrane](https://www.membrane.io/), and [Oxide](https://oxide.computer/) before that. I've known Justin for years, and I'm thrilled to have him on the team.
+
+As per usual, the highlight of our newsletter is at the bottom: a list of the most interesting community vals we've seen this month, including a Val Town clone built on Val Town, Stripe examples, Bluesky vals, virtual money, Cerebras demos, and almost a hundred more!
+
+## 🤖 Townie upgrades
+
+Townie has been massively upgraded this month. We shipped some of the most common feature request and fixed the worst bugs.
+
+TODO video of restoring an old version, and possibly townie detecting and fixing an error
+
+- By far the most common feature request for Townie has been the ability to preview the older versions of HTTP vals. This is now live! There's a dropdown in the bottom of the Preview tab in Townie to select an older version to preview, with the ability to restore that version. Thanks [Charlie](https://www.val.town/u/dcm31) for the tip to add the Restore button, and [Matt](https://www.val.town/u/neverstew) for the feedback on the wording of the button 🙏
+- In order to make Townie sustainable, we rolled out usage limits. Now free-tier users can try out Townie for about a dozen messages on a Haiku, a cheaper (albeit still fantastic) model. We hope this will help you get a sense of what Townie can do, and encourage you to upgrade to Pro if you want to use it regularly. If you find that you're hitting your limits as a Pro user quite often, please reach out to me at steve@val.town with more context on your use case. We're eagerly looking for feedback on a higher-level Pro tier or usage-based pricing.
```

Comment:
```suggestion
- In order to make Townie sustainable, we rolled out usage limits. Now free-tier users can try out Townie for about a dozen messages on a Haiku, a cheaper (albeit still fantastic) model. We hope this will help you get a sense of what Townie can do, and encourage you to upgrade to Pro if you want to use it regularly. If you find that you're hitting your limits as a Pro user, please reach out to me at steve@val.town with more context on your use case. We're eagerly looking for feedback on a higher-level Pro tier or usage-based pricing.
```